dvc/command/diff.py

```diff
@@ -0,0 +1,137 @@
+
+    def _show(self, diff_dct):
+        engine = inflect.engine()
+        msg = "dvc diff from {} to {}".format(
+            diff_dct[git.DIFF_A_REF], diff_dct[git.DIFF_B_REF]
+        )
+        if diff_dct.get(git.DIFF_EQUAL):
+            logger.info(msg)
+            return
+        for dct in diff_dct[git.DIFF_LIST]:
+            msg += "\n\ndiff for '{}' \n".format(dct[git.DIFF_TARGET])
+            if dct.get(git.DIFF_OLD_FILE):
+                msg += "-{} with md5 {}\n".format(
+                    dct[git.DIFF_OLD_FILE], dct[git.DIFF_OLD_CHECKSUM]
+                )
+            if dct.get(git.DIFF_NEW_FILE):
+                msg += "+{} with md5 {}\n".format(
+                    dct[git.DIFF_NEW_FILE], dct[git.DIFF_NEW_CHECKSUM]
+                )
+            msg += "\n"
+            if dct[git.DIFF_SIZE] != git.DIFF_SIZE_UNKNOWN:
+                if dct.get("is_dir"):
+                    msg += "{} {} not changed, ".format(
+                        dct[git.DIFF_IDENT],
+                        engine.plural("file", dct[git.DIFF_IDENT]),
+                    )
+                    msg += "{} {} modified, ".format(
+                        dct[git.DIFF_CHANGE],
+                        engine.plural("file", dct[git.DIFF_CHANGE]),
+                    )
+                    msg += "{} {} added, ".format(
+                        dct[git.DIFF_NEW],
+                        engine.plural("file", dct[git.DIFF_NEW]),
+                    )
+                    msg += "{} {} deleted, ".format(
+                        dct[git.DIFF_DEL],
+                        engine.plural("file", dct[git.DIFF_DEL]),
+                    )
+                    msg += "size was {}".format(
+                        self._print_size(dct[git.DIFF_SIZE])
+                    )
+                else:
+                    if (
+                        dct.get(git.DIFF_OLD_FILE)
+                        and dct.get(git.DIFF_NEW_FILE)
+                        and dct[git.DIFF_SIZE] == 0
+                    ):
+                        msg += "file size was not changed"
+                    elif dct.get(git.DIFF_NEW_FILE):
+                        msg += "added file with size {}".format(
+                            humanize.naturalsize(dct[git.DIFF_SIZE])
+                        )
+                    elif dct.get(git.DIFF_OLD_FILE):
+                        msg += "deleted file with size {}".format(
+                            humanize.naturalsize(abs(dct[git.DIFF_SIZE]))
+                        )
+                    else:
+                        msg += "file was modified, file size {}".format(
+                            self._print_size(dct[git.DIFF_SIZE])
+                        )
+            else:
+                msg += "size is ?"
+        logger.info(msg)
+
+    def run(self):
+        try:
+            msg = self.repo.diff(
+                self.args.target, a_ref=self.args.a_ref, b_ref=self.args.b_ref
```

Comment:
`repo.diff` API doesn't match `dvc diff` CLI, where `target` is optional and `a_ref` is required.